### ISSUE TITLE
Feature: Add ability to extend clean-tag blacklist by custom props

### ISSUE THREAD

**Issue by selrond** [jxnblk/styled-system#150]
Would it be possible to extend `clean-tag`'s blacklist with my custom prop / alias?

For example - I try to add `backgroundRepeat` prop function:

```js
const backgroundRepeat = responsiveStyle({
	prop: 'backgroundRepeat',
	alias: 'bgRepeat'
})
```

but since `bgRepeat` is not part of `clean-tag`s filtering blacklist, I'm getting 

```
Warning: React does not recognize the `bgRepeat` prop on a DOM element. ...
```

I know I could manually omit problematic props, but there has to be a better way - is there?

**Comment by jxnblk** [comment]
You can pass your own array as the `blacklist` prop, but you'll need to include any other props you'd like to include along with it (i.e. it doesn't concat the default array)

**Comment by selrond** [comment]
@jxnblk so wouldn't it make sense to improve this behavior?

**Comment by jxnblk** [comment]
Perhaps. Not tested, but I imagine something like this would work:

```js
import styled from 'styled-components'
import tag from 'clean-tag'
import { backgroundImage, responsiveStyle } from 'styled-system'

const backgroundRepeat = responsiveStyle({
  prop: 'backgroundRepeat',
  alias: 'bgRepeat'
})

tag.defaultProps.blacklist = [
  ...tag.defaultProps.blacklist,
  'backgroundRepeat',
  'bgRepeat'
]

const Foo = styled.div`
  ${backgroundRepeat}
  ${backgroundImage}
`
```

Not opposed to something like a `customBlacklist` prop, but would want to figure out a good name for it... maybe it should accept a function?

**Comment by samtietjen** [comment]
I also ran into this with system components. My work around for right now is:
```javascript
Button.defaultProps.blacklist = [   
  ...Object.keys(Button.propTypes),   
  'buttonSize'   
]
```
Which works, but can pollute props a bit. Is there a reason why blacklist isn't forced to concat with the propType keys?

**Comment by ckeeney** [comment]
Here is a use case for this feature:

```js
import {Text} from 'rebass'

const MyText = Text.extend.attrs({blacklist: ['uppercase']})`
  text-transform: ${props => props.uppercase ? 'uppercase' : 'initial'};
`;
```

With the above code, the blacklist defined by Rebass is overwritten.

**Comment by Kikobeats** [comment]
This definitely should be added into the doc!

**Comment by jxnblk** [comment]
I think this should be resolved with #119
